### PR TITLE
fix: resolve deprecated weight_norm in torch

### LINKED ISSUE #3074: FutureWarning: `torch.nn.utils.weight_norm` is deprecated in favor of `torch.nn.utils.parametrizations.weight_norm`.
### Describe the bug

https://github.com/search?q=repo%3Aspeechbrain%2Fspeechbrain+weight_norm&type=code

I have searched the weight_norm, they are all using the torch.nn.utils.weight_norm

this is the warning in torch

```
    .. warning::

        This function is deprecated.  Use :func:`torch.nn.utils.parametrizations.weight_norm`
        which uses the modern parametrization API.  The new ``weight_norm`` is compatible
        with ``state_dict`` generated from old ``weight_norm``.

        Migration guide:

        * The magnitude (``weight_g``) and direction (``weight_v``) are now expressed
          as ``parametrizations.weight.original0`` and ``parametrizations.weight.original1``
          respectively.  If this is bothering you, please comment on
          https://github.com/pytorch/pytorch/issues/102999

        * To remove the weight normalization reparametrization, use
          :func:`torch.nn.utils.parametrize.remove_parametrizations`.

        * The weight is no longer recomputed once at module forward; instead, it will
          be recomputed on every access.  To restore the old behavior, use
          :func:`torch.nn.utils.parametrize.cached` before invoking the module
          in question.
```

Should it be changed to parametrizations.weight_norm?

### Expected behaviour

Can we solve this warning?

### To Reproduce

_No response_

### Environment Details

_No response_

### Relevant Log Output

```shell

```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Decodetalkers**
resolve: https://github.com/speechbrain/speechbrain/issues/3074

This pr induced a file named torch_weight_norm.py, allow to try import the newest one, if failed , fallback to the old one. This logic was from dac.py, and I made it a separated python file

## What does this PR do?

Fixes https://github.com/speechbrain/speechbrain/issues/3074

<details>
  <summary><b>Before submitting</b></summary>

- [x] Did you read the [contributor guideline](https://speechbrain.readthedocs.io/en/latest/contributing.html)?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing [tests](https://github.com/speechbrain/speechbrain/tree/develop/tests) pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Does your code adhere to project-specific code style and conventions?

</details>

## PR review

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
- [x] Confirm that the changes adhere to compatibility requirements (e.g., Python version, platform)
- [x] Review the self-review checklist to ensure the code is ready for review

</details>

### CODE REVIEW

**Comment by Decodetalkers** [comment]
Ok, this place need to use lazy import.. so I need to copy the code for many times..

**Comment by Decodetalkers** [comment]
Is it because the pretrained model itself is using the old torch?
